### PR TITLE
Make absolute path so that it is more safer to use fsspec

### PR DESCRIPTION
**Pull request by hitchhicker**

### CODE REVIEW

File: autofaiss/external/quantize.py [hunk 1]
```diff
@@ -114,6 +114,10 @@ def build_index(
         Folder to save the temporary small indices that are generated by each spark executor.
         Only used when distributed = "pyspark".
     """
+    index_path = make_path_absolute(index_path)
```
Comment (rom1504) [inline autofaiss/external/quantize.py:117]: this can be done (in case we don't want to save the index), can you check it ?

Comment (hitchhicker) [reply autofaiss/external/quantize.py:117]: I just checked, it works when `save_on_disk` is False. But I think that we should make `index_path` as an optional argument explicitly. (the default value is kept, i.e. knn.index)
I will do another commit to:

- avoid using `make_path_absolute` if `index_path` is None
- raise an valueerror when `index_path` is None but `save_on_disk` is True

Comment (hitchhicker) [reply autofaiss/external/quantize.py]: Ok, I eventually print the error for the above 2nd case and return None as you have done for other edge cases.

File: autofaiss/external/quantize.py [hunk 2]
```diff
@@ -114,6 +114,14 @@ def build_index(
         Folder to save the temporary small indices that are generated by each spark executor.
         Only used when distributed = "pyspark".
     """
+    if index_path is not None:
+        index_path = make_path_absolute(index_path)
+    elif save_on_disk:
+        print("Please specify a index_path if you set save_on_disk as True")
+        return None, None
+    index_infos_path = make_path_absolute(index_infos_path)
```
Comment (rom1504) [inline autofaiss/external/quantize.py:122]: I think this one can also be None, no ?

Comment (hitchhicker) [reply autofaiss/external/quantize.py]: Yeah, indeed. Good catch !

**Pull request merged**